tic_tac_toe/game.py

```diff
@@ -87,18 +136,57 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
+    board = game['board']
+    if game['winner'] or _board_is_full(board):
+        raise InvalidMovement("Game is over.")
+        
+    elif not _position_is_valid(position):
+        raise InvalidMovement('Position out of range.')
+    
+    elif not _position_is_empty_in_board(position, board):
+        raise InvalidMovement('Position already taken.')
+    
+    elif get_next_turn(game) != player:
+        raise InvalidMovement('"{}" moves next'.format(game['next_turn']))
+    
+    if player == game['player1']:
+        game['next_turn'] = game['player2']
+    else:
+        game['next_turn'] = game['player1']
+    
+    board[position[0]][position[1]] = player
+    
+    winner = _check_winning_combinations(board, player)
+    
+    if winner:
+        game['winner'] = winner
```

Comment:
Instead of setting `winner` and then assigning that to `game['winner']` a couple of lines later you could just do `game['winner'] = _check_winning_combinations(board, player)` This works because `_check_winning_combinations` returns `None` if there isn't a winner.